corehq/motech/openmrs/repeater_helpers.py

```diff
@@ -554,14 +554,63 @@ def get_patient_id_type_uuids_values():
     submit_case_blocks([case_block.as_string()], case.domain, xmlns=XMLNS_OPENMRS)
 
 
-def find_patient(requests, domain, case_id, openmrs_config):
-    case = CaseAccessors(domain).get_case(case_id)
+def create_patient(requests, info, case_config):
+    name = {
+        property_: value_source.get_value(info)
+        for property_, value_source in case_config.person_preferred_name.items()
+        if property_ in NAME_PROPERTIES and value_source.get_value(info)
+    }
+    address = {
+        property_: value_source.get_value(info)
+        for property_, value_source in case_config.person_preferred_address.items()
+        if property_ in ADDRESS_PROPERTIES and value_source.get_value(info)
+    }
+    properties = {
+        property_: value_source.get_value(info)
+        for property_, value_source in case_config.person_properties.items()
+        if property_ in PERSON_PROPERTIES and value_source.get_value(info)
+    }
+    person = {}
+    if name:
+        person['names'] = [serialize(name)]
+    if address:
+        person['addresses'] = [serialize(address)]
+    if properties:
+        person.update(serialize(properties))
+    if person:
+        identifiers = [
+            {'identifierType': patient_identifier_type, 'identifier': value_source.get_value(info)}
+            for patient_identifier_type, value_source in case_config.patient_identifiers.items()
+            if patient_identifier_type != PERSON_UUID_IDENTIFIER_TYPE_ID and value_source.get_value(info)
+        ]
+        patient = {
+            'person': person,
+        }
+        if identifiers:
+            patient['identifiers'] = identifiers
+        response = requests.post(
+            '/ws/rest/v1/patient/',
+            json=patient,
+        )
+        if 200 <= response.status_code < 300:
+            # response.json() is not the full patient record. We need
+            # the patient's identifiers and attributes.
+            return get_patient_by_uuid(requests, response.json()['uuid'])
```

Comment:
curious, why do you pass `requests` around? Isn't that an external library?

Reply:
It's an instance of [corehq.motech.requests.Requests](https://github.com/dimagi/commcare-hq/blob/6a382796aa76521b21d5251ea914d5cb8b2ccfce/corehq/motech/requests.py#L45). It has some of the same methods as the library (`get`, `post`, and `delete`) but it does some extra stuff for you, like work out the full URL, authenticate the request, log the response, etc.